### DISCUSSION TITLE
Not all issues recorded in baseline

### DISCUSSION THREAD

**Discussion by WolandPL** [detekt/detekt#4555] [Q&A] [answered]
What could be the reason of not all issues (e.g. UnsafeCallOnNullableType) recorded in baseline?

Baseline creation task:
```
task detektProjectBaseline(type: DetektCreateBaselineTask) {
    description = "Overrides current baseline."
    ignoreFailures.set(true)
    parallel.set(true)
    buildUponDefaultConfig.set(true)
    setSource(files(rootDir))
    config.setFrom(configFile)
    baseline.set(baselineFile)
    include(kotlinFiles)
    exclude(resourceFiles, buildFiles)
}
```

And Gradle configuration of modules' detekt tasks:
```
detekt {
        parallel = true
        buildUponDefaultConfig = true
        config = configFile
        baseline = baselineFile
    }
```

**Comment by WolandPL** [comment, marked as answer]
I'm now wondering - it seems that rules that require type resolution are not recorded in the baseline - isn't that the case?

**Reply by cortinico** [reply]
> I'm now wondering - it seems that rules that require type resolution are not recorded in the baseline - isn't that the case?

The Baseline task is nothing more than a regular Detekt task that is writing output on a file. Therefore if you're running the baseline without Type Resolution you'll see fewer findings. 

That's exactly the case in your scenario, as you're using a custom task (why can't you used the Detekt Gradle Plugin?).

**Reply by WolandPL** [reply]
> why can't you used the Detekt Gradle Plugin).

I'm following the docs recommendation for having single baseline for multi-module project (https://detekt.dev/baseline.html)

**Reply by WolandPL** [reply]
I was playing with the Gradle Plugin a bit and you're right, that should work fine. Thank you!
